examples/mycoind/app/app_test.go

```diff
@@ -33,15 +34,16 @@ func testInitChain(t *testing.T, myApp app.BaseApp, addr string) {
                 }]
             }]}`, addr)
 	assert.Equal(t, "", myApp.GetChainID())
-	myApp.InitChain(abci.RequestInitChain{AppStateBytes: []byte(appState)})
+	myApp.InitChain(abci.RequestInitChain{AppStateBytes: []byte(appState), ChainId: chainID})
+	assert.Equal(t, chainID, myApp.GetChainID())
+
 }
 
 // testCommit will commit at height h and return new hash
-func testCommit(t *testing.T, myApp app.BaseApp, h int64, chainId string) []byte {
+func testCommit(t *testing.T, myApp app.BaseApp, h int64) []byte {
 	// Commit first block, make sure non-nil hash
-	header := abci.Header{Height: h, ChainID: chainId}
+	header := abci.Header{Height: h}
```

Comment:
Better this way.
Just wondering, maybe we verify chainID of every header on BeginBlock? And return an error if it is present and different from the one stored in InitCHain

Reply:
Well, I was not sure it's necessary, but we can indeed panic with an error if stuff does not match. Kind of provides "extra security", but on the other hand this should not be happening if the init was correct, otherwise we can't trust tendermint at all. Unless I'm missing something.

What do you think? 

I guess what I'm trying to say is: double checking is good, but it leads to more code and more things that could potentially break/need to be supported while sometimes providing us with no benefit. I used to do that a lot on every level of an application, but then learned to limit myself to what's necessary. 

I might be wrong about this one, so I leave this call up to you.

Reply:
Makes me look ridiculous, fighting for one line of code here 🗡

Reply:
Please leave the tests as is.

I just got contacted from some people, where they ended up replaying blocks from the wrong chain in some dev setup. I thought a second check would not be bad.

But on reflection, the Info method on handshake will show a different AppHash than the chain, so tendermint should prohibit that anyway.

Ignore this comment.

Reply:
@ethanfrey cool, not changing anything here then.